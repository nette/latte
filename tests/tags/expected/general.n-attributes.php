<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	public const Blocks = [
		['bl' => 'blockBl'],
	];


	public function main(array $ʟ_args): void
	{
%A%
		echo '
<p';
		$ʟ_tmp = ['title' => 'hello', 'lang' => isset($lang) ? $lang : null];
		echo Latte\Essential\Nodes\NAttrNode::attrs(isset($ʟ_tmp[0]) && is_array($ʟ_tmp[0]) ? $ʟ_tmp[0] : $ʟ_tmp, false) /* line %d% */;
		echo '> </p>

<p';
		$ʟ_tmp = [['title' => 'hello']];
		echo Latte\Essential\Nodes\NAttrNode::attrs(isset($ʟ_tmp[0]) && is_array($ʟ_tmp[0]) ? $ʟ_tmp[0] : $ʟ_tmp, false) /* line %d% */;
		echo '> </p>

';
		foreach ($iterator = $ʟ_it = new Latte\Essential\CachingIterator([1, 2, 3], $ʟ_it ?? null) as $foo) /* line %d% */ {
			echo '	<b';
			echo ($ʟ_tmp = array_filter([$iterator->even ? 'even' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>item</b>
';

		}
		$iterator = $ʟ_it = $ʟ_it->getParent();

		echo "\n";
		foreach ($iterator = $ʟ_it = new Latte\Essential\CachingIterator([1, 2, 3], $ʟ_it ?? null) as $foo) /* line %d% */ {
			echo '<p';
			echo ($ʟ_tmp = array_filter([$iterator->even ? 'even' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>';
			echo LR\Filters::escapeHtmlText($foo) /* line %d% */;
			echo '</p>
';

		}
		$iterator = $ʟ_it = $ʟ_it->getParent();

		echo '
<p';
		echo ($ʟ_tmp = array_filter(['foo', false ? 'first' : null, 'odd', true ? 'foo' : 'bar'])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class</p>

<p';
		echo ($ʟ_tmp = array_filter([false ? 'first' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class empty</p>
<p';
		echo ($ʟ_tmp = array_filter([true ? 'bem--modifier' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class with BEM</p>


';
		$this->renderBlock('bl', get_defined_vars()) /* line %d% */;
		echo '




<ul title="foreach">
';
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';

		}

		echo '</ul>

<ul title="foreach">
';
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>
		';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '
	</li>
';

		}

		echo '</ul>

<ul title="inner foreach">
	<li>
';
		foreach ($people as $person) /* line %d% */ {
			echo '		';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo "\n";

		}

		echo '	</li>
</ul>

<ul title="tag if">
	';
		$ʟ_tag[0] = '';
		if (true) /* line %d% */ {
			echo '<';
			echo $ʟ_tmp = ('li');
			$ʟ_tag[0] = '</' . $ʟ_tmp . '>' . $ʟ_tag[0];
			echo '>';
		}
		echo '
		';
		echo LR\Filters::escapeHtmlText($person) /* line %d% */;
		echo '
	';
		echo $ʟ_tag[0];
		echo '
</ul>

<ul title="for">
';
		for ($i = 0;
		$i < 3;
		$i++) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($i) /* line %d% */;
			echo '</li>
';

		}
		echo '</ul>

<ul title="white">
';
		while (--$i > 0) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($i) /* line %d% */;
			echo '</li>
';

		}
		echo '</ul>

';
		if (true) /* line %d% */ {
			echo '<p>
	<div><p>true</div>
</p>
';
		}
		echo "\n";
		if (true) /* line %d% */ {
			echo '<p>
	<div><p>true</p></div>
</p>
';
		}
		echo "\n";
		if (false) /* line %d% */ {
			echo '<p>
	<div><p>false</div>
</p>
';
		}
		echo "\n";
		if (false) /* line %d% */ {
			echo '<p>
	<div><p>false</p></div>
</p>
';
		}
		echo "\n";
		if (strlen('{$name}') > 5) /* line %d% */ {
			echo '<p>noLatte</p>
';
		}
		echo '
<ul title="if + foreach">
';
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person) === 4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				echo '</li>
';
			}

		}

		echo '</ul>

<ul title="if + inner-if + inner-foreach">
';
		if (empty($iterator)) /* line %d% */ {
			echo '	<li>';
			foreach ($people as $person) /* line %d% */ {
				if (strlen($person) === 4) /* line %d% */ {
					echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				}

			}

			echo '</li>
';
		}
		echo '</ul>

<ul title="inner-if + inner-foreach">
';
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person) === 4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText(($this->filters->lower)($person)) /* line %d% */;
				echo '</li>
';
			}

		}

		echo '</ul>

';
		$ʟ_tag[1] = '';
		if (true) /* line %d% */ {
			echo '<';
			echo $ʟ_tmp = ('b');
			$ʟ_tag[1] = '</' . $ʟ_tmp . '>' . $ʟ_tag[1];
			echo '>';
		}
		echo 'bold';
		echo $ʟ_tag[1];
		echo ' ';
		$ʟ_tag[2] = '';
		if (false) /* line %d% */ {
			echo '<';
			echo $ʟ_tmp = ('b');
			$ʟ_tag[2] = '</' . $ʟ_tmp . '>' . $ʟ_tag[2];
			echo '>';
		}
		echo 'normal';
		echo $ʟ_tag[2];
		echo '

';
		$ʟ_tag[3] = '';
		if (true) /* line %d% */ {
			echo '<';
			echo $ʟ_tmp = ('b');
			$ʟ_tag[3] = '</' . $ʟ_tmp . '>' . $ʟ_tag[3];
			echo ($ʟ_tmp = array_filter(['first'])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>';
		}
		echo 'bold';
		echo $ʟ_tag[3];
		echo '

<meta>
';
		if (true) /* line %d% */ {
			echo '<meta>
';
		}
		echo '<meta>

';
		foreach ([0] as $foo) /* line %d% */ {
			if (1) /* line %d% */ {
				$ʟ_tag[4] = '';
				foreach ([1] as $foo) /* line %d% */ {
					if (0) /* line %d% */ {
						echo '<';
						echo $ʟ_tmp = ('span');
						$ʟ_tag[4] = '</' . $ʟ_tmp . '>' . $ʟ_tag[4];
						echo '>';
					}

				}

				foreach ([2] as $foo) /* line %d% */ {
					if (2) /* line %d% */ {
						echo 'Hello';
					}

				}

				echo $ʟ_tag[4];
				echo "\n";
			}

		}

		echo '

';
		$ʟ_tag[5] = '';
		if (true) /* line %d% */ {
			echo '<';
			echo $ʟ_tmp = ('div');
			$ʟ_tag[5] = '</' . $ʟ_tmp . '>' . $ʟ_tag[5];
			echo '>';
		}
		echo "\n";
		$ʟ_try[6] = [$ʟ_it ?? null];
		ob_start(fn() => '');
		try /* line %d% */ {
			echo '	';
			$ʟ_tag[7] = '';
			if (false) /* line %d% */ {
				echo '<';
				echo $ʟ_tmp = ('span');
				$ʟ_tag[7] = '</' . $ʟ_tmp . '>' . $ʟ_tag[7];
				echo '>';
			}
			echo "\n";
			throw new Latte\Essential\RollbackException;
			echo '	';
			echo $ʟ_tag[7];
			echo "\n";

		} catch (Throwable $ʟ_e) {
			ob_end_clean();
			if (!($ʟ_e instanceof Latte\Essential\RollbackException) && isset($this->global->coreExceptionHandler)) {
				($this->global->coreExceptionHandler)($ʟ_e, $this);
			}
			ob_start();
		} finally {
			echo ob_get_clean();
			$iterator = $ʟ_it = $ʟ_try[6][0];
		}echo $ʟ_tag[5];
		echo '


<ul title="foreach break">
';
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ {
				echo "</li>\n";
				break;
				;
			}
			echo '</li>
';

		}

		echo '</ul>

<ul title="foreach continue">
';
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ {
				echo "</li>\n";
				continue;
				;
			}
			echo '</li>
';

		}

		echo '</ul>


<ul title="inner foreach break">
	<li>';
		foreach ($people as $person) /* line %d% */ {
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ break;

		}

		echo '</li>
</ul>

<ul title="inner foreach continue">
	<li>';
		foreach ($people as $person) /* line %d% */ {
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ continue;

		}

		echo '</li>
</ul>


';
		ob_start(fn() => '');
		try {
			$ʟ_tag[8] = '';
			echo '<';
			echo $ʟ_tmp = (Latte\Essential\Nodes\NTagNode::check('div', 'span'));
			$ʟ_tag[8] = '</' . $ʟ_tmp . '>' . $ʟ_tag[8];
			echo '>';
			ob_start();
			try {
				echo 'n:tag & n:ifcontent';

			} finally {
				$ʟ_ifc[0] = rtrim(ob_get_flush()) === '';
			}
			echo $ʟ_tag[8];
			echo "\n";

		} finally {
			if ($ʟ_ifc[0] ?? null) {
				ob_end_clean();
			} else {
				echo ob_get_clean();
			}
		}
		ob_start(fn() => '');
		try {
			$ʟ_tag[9] = '';
			echo '<';
			echo $ʟ_tmp = (Latte\Essential\Nodes\NTagNode::check('div', 'span'));
			$ʟ_tag[9] = '</' . $ʟ_tmp . '>' . $ʟ_tag[9];
			echo '>';
			ob_start();
			try {

			} finally {
				$ʟ_ifc[1] = rtrim(ob_get_flush()) === '';
			}
			echo $ʟ_tag[9];
			echo "\n";

		} finally {
			if ($ʟ_ifc[1] ?? null) {
				ob_end_clean();
			} else {
				echo ob_get_clean();
			}
		}
		echo "\n";
		if (1) /* line %d% */ {
			$ʟ_tag[10] = '';
			echo '<';
			echo $ʟ_tmp = (Latte\Essential\Nodes\NTagNode::check('div', 'span'));
			$ʟ_tag[10] = '</' . $ʟ_tmp . '>' . $ʟ_tag[10];
			echo '>n:tag & n:if=1';
			echo $ʟ_tag[10];
			echo "\n";
		}
		if (0) /* line %d% */ {
			$ʟ_tag[11] = '';
			echo '<';
			echo $ʟ_tmp = (Latte\Essential\Nodes\NTagNode::check('div', 'span'));
			$ʟ_tag[11] = '</' . $ʟ_tmp . '>' . $ʟ_tag[11];
			echo '>n:tag & n:if=0';
			echo $ʟ_tag[11];
			echo "\n";
		}
	}


	public function prepare(): array
	{
		extract($this->params);

		if (!$this->getReferringTemplate() || $this->getReferenceType() === 'extends') {
			foreach (array_intersect_key(['foo' => '6, 10, 94, 94, 94', 'person' => '19, 27, 31, 37, 75, 79, 82, 107, 111, 116, 120'], $this->params) as $ʟ_v => $ʟ_l) {
				trigger_error("Variable \$$ʟ_v overwritten in foreach on line $ʟ_l");
			}
		}
		return get_defined_vars();
	}


	/** n:block="bl" on line %d% */
	public function blockBl(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);

		echo '<ul title="block + if + foreach">
';
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person) === 4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				echo '</li>
';
			}

		}

		echo '</ul>
';
	}
}
