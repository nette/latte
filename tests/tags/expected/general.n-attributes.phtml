<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['bl' => 'blockBl'],
	];


	public function main(): array
	{
%A%
		echo '
<p';
		$ʟ_tmp = ['title' => 'hello', 'lang' => isset($lang) ? $lang : null];
		echo LR\Filters::htmlAttributes(isset($ʟ_tmp[0]) && is_array($ʟ_tmp[0]) ? $ʟ_tmp[0] : $ʟ_tmp) /* line %d% */;
		echo '> </p>

<p';
		$ʟ_tmp = [['title' => 'hello']];
		echo LR\Filters::htmlAttributes(isset($ʟ_tmp[0]) && is_array($ʟ_tmp[0]) ? $ʟ_tmp[0] : $ʟ_tmp) /* line %d% */;
		echo '> </p>

';
		$iterations = 0;
		foreach ($iterator = $ʟ_it = new LR\CachingIterator([1,2,3], $ʟ_it ?? null) as $foo) /* line %d% */ {
			echo '	<b';
			echo ($ʟ_tmp = array_filter([$iterator->even ? 'even' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>item</b>
';
			$iterations++;
		}
		$iterator = $ʟ_it = $ʟ_it->getParent();
		echo "\n";
		$iterations = 0;
		foreach ($iterator = $ʟ_it = new LR\CachingIterator([1, 2, 3], $ʟ_it ?? null) as $foo) /* line %d% */ {
			echo '<p';
			echo ($ʟ_tmp = array_filter([$iterator->even ? 'even' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>';
			echo LR\Filters::escapeHtmlText($foo) /* line %d% */;
			echo '</p>
';
			$iterations++;
		}
		$iterator = $ʟ_it = $ʟ_it->getParent();
		echo '
<p';
		echo ($ʟ_tmp = array_filter(['foo', (false ? 'first' : null), 'odd', (true ? 'foo' : 'bar')])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class</p>

<p';
		echo ($ʟ_tmp = array_filter([false ? 'first' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class empty</p>
<p';
		echo ($ʟ_tmp = array_filter([true ? 'bem--modifier' : null])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
		echo '>n:class with BEM</p>


';
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		$this->renderBlock('bl', get_defined_vars()) /* line %d% */;
		echo '




<ul title="foreach">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '</li>
';
			$iterations++;
		}
		echo '</ul>

<ul title="foreach">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>
		';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo '
	</li>
';
			$iterations++;
		}
		echo '</ul>

<ul title="inner foreach">
	<li>
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '		';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			echo "\n";
			$iterations++;
		}
		echo '	</li>
</ul>

<ul title="tag if">
';
		if ($ʟ_if[1] = (true)) /* line %d% */ {
			echo '	<li>
';
		}
		echo '		';
		echo LR\Filters::escapeHtmlText($person) /* line %d% */;
		echo "\n";
		if ($ʟ_if[1]) /* line %d% */ {
			echo '	</li>
';
		}
		echo '</ul>

<ul title="for">
';
		for ($i=0;
		$i<3;
		$i++) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($i) /* line %d% */;
			echo '</li>
';
		}
		echo '</ul>

<ul title="white">
';
		while (--$i>0) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($i) /* line %d% */;
			echo '</li>
';
		}
		echo '</ul>

';
		if (true) /* line %d% */ {
			echo '<p>
	<div><p>true</div>
</p>
';
		}
		echo "\n";
		if (true) /* line %d% */ {
			echo '<p>
	<div><p>true</p></div>
</p>
';
		}
		echo "\n";
		if (false) /* line %d% */ {
			echo '<p>
	<div><p>false</div>
</p>
';
		}
		echo "\n";
		if (false) /* line %d% */ {
			echo '<p>
	<div><p>false</p></div>
</p>
';
		}
		echo "\n";
		if (strlen('{$name}') > 5) /* line %d% */ {
			echo '<p>noLatte</p>
';
		}
		echo '
<ul title="if + foreach">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person)===4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				echo '</li>
';
			}
			$iterations++;
		}
		echo '</ul>

<ul title="if + inner-if + inner-foreach">
';
		if (empty($iterator)) /* line %d% */ {
			echo '	<li>';
			$iterations = 0;
			foreach ($people as $person) /* line %d% */ {
				if (strlen($person)===4) /* line %d% */ {
					echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				}
				$iterations++;
			}
			echo '</li>
';
		}
		echo '</ul>

<ul title="inner-if + inner-foreach">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person)===4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText(($this->filters->lower)($person)) /* line %d% */;
				echo '</li>
';
			}
			$iterations++;
		}
		echo '</ul>

';
		if ($ʟ_if[0] = (true)) /* line %d% */ {
			echo '<b>';
		}
		echo 'bold';
		if ($ʟ_if[0]) /* line %d% */ {
			echo '</b>';
		}
		echo ' ';
		if ($ʟ_if[0] = (false)) /* line %d% */ {
			echo '<b>';
		}
		echo 'normal';
		if ($ʟ_if[0]) /* line %d% */ {
			echo '</b>
';
		}
		echo "\n";
		if ($ʟ_if[0] = (true)) /* line %d% */ {
			echo '<b';
			echo ($ʟ_tmp = array_filter(['first'])) ? ' class="' . LR\Filters::escapeHtmlAttr(implode(" ", array_unique($ʟ_tmp))) . '"' : "" /* line %d% */;
			echo '>';
		}
		echo 'bold';
		if ($ʟ_if[0]) /* line %d% */ {
			echo '</b>
';
		}
		echo '
<meta>
';
		if (true) /* line %d% */ {
			echo '<meta>
';
		}
		echo '<meta>

';
		$iterations = 0;
		foreach ([0] as $foo) /* line %d% */ {
			if (1) /* line %d% */ {
				$iterations = 0;
				foreach ([1] as $foo) /* line %d% */ {
					if ($ʟ_if[0] = (0)) /* line %d% */ {
						echo '<span>';
					}
					$iterations++;
				}
				$iterations = 0;
				foreach ([2] as $foo) /* line %d% */ {
					if (2) /* line %d% */ {
						echo 'Hello';
					}
					$iterations++;
				}
				$iterations = 0;
				foreach ([1] as $foo) /* line %d% */ {
					if ($ʟ_if[0]) /* line %d% */ {
						echo '</span>
';
					}
					$iterations++;
				}
			}
			$iterations++;
		}
		echo '

';
		if ($ʟ_if[0] = (true)) /* line %d% */ {
			echo '<div>
';
		}
		$ʟ_try[0] = [$ʟ_it ?? null];
		ob_start(function () {});
		try /* line %d% */ {
			if ($ʟ_if[1] = (false)) /* line %d% */ {
				echo '	<span>
';
			}
			throw new LR\RollbackException;
			if ($ʟ_if[1]) /* line %d% */ {
				echo '	</span>
';
			}

		} catch (Throwable $ʟ_e) {
			ob_end_clean();
			if (!($ʟ_e instanceof LR\RollbackException) && isset($this->global->coreExceptionHandler)) {
				($this->global->coreExceptionHandler)($ʟ_e, $this);
			}

			ob_start();
		} finally {
			echo ob_get_clean();
			$iterator = $ʟ_it = $ʟ_try[0][0];
		}
		if ($ʟ_if[0]) /* line %d% */ {
			echo '</div>
';
		}
		echo '

<ul title="foreach break">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ {
				echo "</li>\n";
				break;
			}
			echo '</li>
';
			$iterations++;
		}
		echo '</ul>

<ul title="foreach continue">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo '	<li>';
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ {
				echo "</li>\n";
				continue;
			}
			echo '</li>
';
			$iterations++;
		}
		echo '</ul>


<ul title="inner foreach break">
	<li>';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ break;
			$iterations++;
		}
		echo '</li>
</ul>

<ul title="inner foreach continue">
	<li>';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			echo LR\Filters::escapeHtmlText($person) /* line %d% */;
			if (true) /* line %d% */ continue;
			$iterations++;
		}
		echo '</li>
</ul>
';
%A%
	}


	public function prepare(): void
	{
%A%
	}


	/** {block bl} on line %d% */
	public function blockBl(array $ʟ_args): void
	{
		extract($this->params);
		extract($ʟ_args);
		unset($ʟ_args);
		echo '<ul title="block + if + foreach">
';
		$iterations = 0;
		foreach ($people as $person) /* line %d% */ {
			if (strlen($person)===4) /* line %d% */ {
				echo '	<li>';
				echo LR\Filters::escapeHtmlText($person) /* line %d% */;
				echo '</li>
';
			}
			$iterations++;
		}
		echo '</ul>
';
	}

}
