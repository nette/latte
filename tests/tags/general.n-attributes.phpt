<?php

/**
 * Test: general n:attributes test.
 */

declare(strict_types=1);

use Tester\Assert;

require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;

$params['people'] = ['John', 'Mary', 'Paul'];

Assert::matchFile(
	__DIR__ . '/expected/general.n-attributes.php',
	$latte->compile(__DIR__ . '/templates/n-attributes.latte'),
);
Assert::matchFile(
	__DIR__ . '/expected/general.n-attributes.html',
	$latte->renderToString(
		__DIR__ . '/templates/n-attributes.latte',
		$params,
	),
);
