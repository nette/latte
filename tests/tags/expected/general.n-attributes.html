
<p title="hello"> </p>

<p title="hello"> </p>

	<b>item</b>
	<b class="even">item</b>
	<b>item</b>

<p>1</p>
<p class="even">2</p>
<p>3</p>

<p class="foo odd">n:class</p>

<p>n:class empty</p>
<p class="bem--modifier">n:class with BEM</p>


<ul title="block + if + foreach">
	<li>John</li>
	<li>Mary</li>
	<li>Paul</li>
</ul>





<ul title="foreach">
	<li>John</li>
	<li>Mary</li>
	<li>Paul</li>
</ul>

<ul title="foreach">
	<li>
		John
	</li>
	<li>
		Mary
	</li>
	<li>
		Paul
	</li>
</ul>

<ul title="inner foreach">
	<li>
		John
		Mary
		Paul
	</li>
</ul>

<ul title="tag if">
	<li>
		Paul
	</li>
</ul>

<ul title="for">
	<li>0</li>
	<li>1</li>
	<li>2</li>
</ul>

<ul title="white">
	<li>2</li>
	<li>1</li>
</ul>

<p>
	<div><p>true</div>
</p>

<p>
	<div><p>true</p></div>
</p>



<p>noLatte</p>

<ul title="if + foreach">
	<li>John</li>
	<li>Mary</li>
	<li>Paul</li>
</ul>

<ul title="if + inner-if + inner-foreach">
	<li>JohnMaryPaul</li>
</ul>

<ul title="inner-if + inner-foreach">
	<li>john</li>
	<li>mary</li>
	<li>paul</li>
</ul>

<b>bold</b> normal
<b class="first">bold</b>

<meta>
<meta>
<meta>

Hello

<div>
</div>


<ul title="foreach break">
	<li>John</li>
</ul>

<ul title="foreach continue">
	<li>John</li>
	<li>Mary</li>
	<li>Paul</li>
</ul>


<ul title="inner foreach break">
	<li>John</li>
</ul>

<ul title="inner foreach continue">
	<li>JohnMaryPaul</li>
</ul>
